File: app/src/main/java/com/savvasdalkitsis/gameframe/injector/infra/navigation/NavigatorInjector.java
package com.savvasdalkitsis.gameframe.injector.infra.navigation;

import com.savvasdalkitsis.gameframe.infra.navigation.ApplicationNavigator;
import com.savvasdalkitsis.gameframe.infra.navigation.Navigator;
import com.savvasdalkitsis.gameframe.injector.ApplicationInjector;
import com.savvasdalkitsis.gameframe.injector.infra.TopActivityProviderInjector;

public class NavigatorInjector {

    public static Navigator navigator() {
        return new ApplicationNavigator(TopActivityProviderInjector.topActivityProvider(), ApplicationInjector.application());
    }
}

File: app/src/main/java/com/savvasdalkitsis/gameframe/injector/infra/navigation/NavigatorInjector.kt
package com.savvasdalkitsis.gameframe.injector.infra.navigation

import com.savvasdalkitsis.gameframe.navigation.ApplicationNavigator
import com.savvasdalkitsis.gameframe.navigation.Navigator
import com.savvasdalkitsis.gameframe.injector.ApplicationInjector
import com.savvasdalkitsis.gameframe.injector.infra.TopActivityProviderInjector

object NavigatorInjector {

    fun navigator(): Navigator =
            ApplicationNavigator(TopActivityProviderInjector.topActivityProvider(),
                    ApplicationInjector.application())
}
